scripts/benchmark/checks/deploy_erc20.py

```diff
@@ -0,0 +1,167 @@
+        self.contract_interface = compiled_sol['{}:SimpleToken'.format(SOLIDITY_SRC_FILE)]
+
+        # Stop Web3 from trying to call out to external APIs
+        TRANSACTION_DEFAULTS['gas'] = lambda web3, tx: 0
+        TRANSACTION_DEFAULTS['gasPrice'] = lambda web3, tx: 0
+
+        self.w3 = Web3()
+
+    @property
+    def name(self) -> str:
+        return 'ERC20 deployment'
+
+    def execute(self) -> DefaultStat:
+        total_stat = DefaultStat()
+
+        for chain in get_all_chains():
+
+            value = self.as_timed_result(
+                lambda: self.mine_blocks(chain, self.num_blocks, self.num_tx)
+            )
+
+            total_gas_used, total_num_tx = value.wrapped_value
+
+            stat = DefaultStat(
+                caption=chain.get_vm().fork,
+                total_blocks=self.num_blocks,
+                total_tx=total_num_tx,
+                total_seconds=value.duration,
+                total_gas=total_gas_used,
+            )
+            total_stat = total_stat.cumulate(stat)
+            self.print_stat_line(stat)
+
+        return total_stat
+
+    def mine_blocks(self, chain: Chain, num_blocks: int, num_tx: int) -> Tuple[int, int]:
+        total_gas_used = 0
+        total_num_tx = 0
+
+        for i in range(1, num_blocks + 1):
+            block = self.mine_block(chain, i, num_tx)
+            total_gas_used = total_gas_used + block.header.gas_used
+            total_num_tx = total_num_tx + len(block.transactions)
+
+        return total_gas_used, total_num_tx
+
+    def mine_block(self,
+                   chain: Chain,
+                   block_number: int,
+                   num_tx: int) -> BaseBlock:
+
+        for i in range(1, num_tx + 1):
+            self.apply_transaction(chain)
+
+        return chain.mine_block()
+
+    def apply_transaction(self, chain: Chain) -> None:
+
+        # Instantiate the contract
+        SimpleToken = self.w3.eth.contract(
+            abi=self.contract_interface['abi'],
+            bytecode=self.contract_interface['bin']
+        )
+
+        # Build transaction to deploy the contract
+        w3_tx1 = SimpleToken.constructor().buildTransaction()
+
+        tx = new_transaction(
+            vm=chain.get_vm(),
+            private_key=FUNDED_ADDRESS_PRIVATE_KEY,
+            from_=FUNDED_ADDRESS,
+            to=CREATE_CONTRACT_ADDRESS,
+            amount=0,
+            gas=FIRST_TX_GAS_LIMIT,
+            data=decode_hex(w3_tx1['data']),
+        )
+
+        logging.debug('Applying Transaction {}'.format(tx))
+
+        block, receipt, computation = chain.apply_transaction(tx)
+        deployed_contract_address = computation.msg.storage_address
+
+        assert computation.is_success
+
+        # Interact with the deployed contract by calling the totalSupply() API
+        simple_token = self.w3.eth.contract(
+            address=Web3.toChecksumAddress(encode_hex(deployed_contract_address)),
```

Comment:
In the next release, web3py will accept `bytes` addresses, but it doesn't seem worth a wait (or rushing out the release), for a minor nicety.